test/WebJobs.Extensions.DurableTask.Analyzers.Test/Orchestrator/AwaitAnalyzerTests.cs

```diff
@@ -0,0 +1,208 @@
+                            new DiagnosticResultLocation("Test0.cs", 15, 18)
+                        }
+            };
+
+            expectedDiagnostics[1] = new DiagnosticResult
+            {
+                Id = MethodAnalyzer.DiagnosticId,
+                Message = string.Format(Resources.MethodAnalyzerMessageFormat, "UserDefinedCode()"),
+                Severity = Severity,
+                Locations =
+                    new[] {
+                            new DiagnosticResultLocation("Test0.cs", 15, 24)
+                        }
+            };
+
+            expectedDiagnostics[2] = new DiagnosticResult
+            {
+                Id = DiagnosticId,
+                Message = Resources.AwaitAnalyzerMessageFormat,
+                Severity = Severity,
+                Locations =
+                    new[] {
+                            new DiagnosticResultLocation("Test0.cs", 20, 17)
+                        }
+            };
+
+            VerifyCSharpDiagnostic(test, expectedDiagnostics);
+        }
+
+        // should flag when used in method directly or indirectly called using method analyzer
+        // should flag in orchestrator method
+        [TestMethod]
+        public void AwaitAnalyzer_UserDefinedCode_NonDeterministicMethod()
+        {
+            var test = @"
+    using System;
+    using System.Threading.Tasks;
+    using Microsoft.Azure.WebJobs;
+    using Microsoft.Azure.WebJobs.Extensions.DurableTask;
+
+    namespace VSSample
+    {
+        public static class HelloSequence
+        {
+            [FunctionName(""AwaitAnalyzerTestCases"")]
+            public static async Task Run(
+            [OrchestrationTrigger] IDurableOrchestrationContext context
+            {
+                DirectCall();
+            }
+
+            public static void DirectCall()
+            {
+                await UserDefinedCode();
+            }
+
+            public static Task UserDefinedCode()
+            {
+            }
+        }
+    }";
+
+            var expectedDiagnostics = new DiagnosticResult[2];
+            expectedDiagnostics[0] = new DiagnosticResult
+            {
+                Id = MethodAnalyzer.DiagnosticId,
+                Message = string.Format(Resources.MethodAnalyzerMessageFormat, "DirectCall()"),
+                Severity = Severity,
+                Locations =
+                    new[] {
+                            new DiagnosticResultLocation("Test0.cs", 15, 17)
+                        }
+            };
+
+            expectedDiagnostics[1] = new DiagnosticResult
+            {
+                Id = DiagnosticId,
+                Message = Resources.AwaitAnalyzerMessageFormat,
+                Severity = Severity,
+                Locations =
+                    new[] {
+                            new DiagnosticResultLocation("Test0.cs", 20, 17)
+                        }
+            };
+
+            VerifyCSharpDiagnostic(test, expectedDiagnostics);
+        }
```

Comment:
What happens if I have code like this?
```csharp
[FunctionName("AwaitAnalyzerTestCases")]
public static async Task Run([OrchestrationTrigger] IDurableOrchestrationContext context)
{
    await Method(context);
}

private static async Task Method(IDurableOrchestrationContext context)
{
    await context.CallActivityAsync("Hello", "World");
}
```

Will this pass without a warning? It should, since it doesn't violate any determinism code constraints.